core/src/main/java/org/infinispan/atomic/impl/ApplyDelta.java

```diff
@@ -0,0 +1,93 @@
+package org.infinispan.atomic.impl;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+import org.infinispan.atomic.CopyableDeltaAware;
+import org.infinispan.atomic.Delta;
+import org.infinispan.atomic.DeltaAware;
+import org.infinispan.commons.CacheException;
+import org.infinispan.commons.api.functional.EntryView;
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.Ids;
+import org.infinispan.commons.marshall.Marshaller;
+import org.infinispan.commons.util.Util;
+import org.infinispan.factories.GlobalComponentRegistry;
+
+/**
+ * Replacement for {@link org.infinispan.commands.write.ApplyDeltaCommand} and
+ * {@link org.infinispan.context.Flag#DELTA_WRITE}. Deprecated since {@link Delta},
+ * {@link org.infinispan.atomic.DeltaAware} and {@link org.infinispan.atomic.CopyableDeltaAware} are deprecated, too.
+ * @deprecated since 9.1
+ */
+@Deprecated
+public final class ApplyDelta<K> implements BiFunction<Object, EntryView.ReadWriteEntryView<K, Object>, Object> {
+   private final Marshaller marshaller;
+
+   public ApplyDelta(Marshaller marshaller) {
+      this.marshaller = marshaller;
+   }
+
+   @Override
+   public Object apply(Object d, EntryView.ReadWriteEntryView<K, Object> view) {
+      if (!(d instanceof Delta)) {
+         throw new IllegalArgumentException("Expected delta, argument is " + d);
+      }
+      Delta delta = (Delta) d;
+      if (view.find().isPresent()) {
+         Object value = view.find().get();
+         DeltaAware deltaAware;
+         if (value instanceof CopyableDeltaAware) {
+            deltaAware = ((CopyableDeltaAware) value).copy();
+         } else if (value instanceof DeltaAware) {
+            try {
+               byte[] bytes = marshaller.objectToByteBuffer(value);
```

Comment:
@rvansa Could it use `MergeOnStore` instead? Not that I think `MergeOnStore` is a good idea, but it seems quite close to the semantics of non-copyable `DeltaAware`.